src/mm/checkpoints.c

```diff
@@ -42,78 +42,143 @@
 
 
 
-bool check_not_used_chunk_and_clean(malloc_area *m_area, int *bitmap_blocks) {
+int compute_bitmap_blocks(int chunks) {
+
+	int blocks = chunks / NUM_CHUNKS_PER_BLOCK;
+	if (blocks < 1)
+		blocks = 1;
+
+	return blocks;
+
+}
+
+void set_chunk_dirty_from_address(void *addr, int lp) {
+
+	size_t chunk_size, area_size;
+	int j, k, i, blocks, num_chunks;
+	unsigned long long marea_base_addr, chunk_base_address;
+	malloc_area *target_marea;
+	malloc_state *m_state;
+	unsigned long long offset;
+	unsigned int index;
+
+	unsigned long long start_address = (unsigned long long)addr;
+
+	m_state = recoverable_state[lp];
+	int num_areas = m_state->num_areas;
+
+	malloc_area *areas = m_state->areas;
+	assert(areas != NULL);
+	
+
+	for (j=0; j < num_areas; j++) {
+		num_chunks = areas[j].num_chunks;
+		chunk_size = areas[j].chunk_size;
+		blocks = compute_bitmap_blocks(num_chunks);
+		area_size = num_chunks * chunk_size;
+		marea_base_addr = ((unsigned long long) areas[j].area); 
+		if (start_address >= marea_base_addr && start_address < marea_base_addr + area_size) {
+			target_marea = &areas[j];
+			offset = start_address - marea_base_addr;
+			index = offset / chunk_size;
+			SET_DIRTY_BIT(target_marea, index); 
+			break;
+		}
+	}
+
+	
+}
+
+void clean_bitmap(unsigned int *use_bitmap, int blocks, unsigned int **bitmap) {
```

Comment:
why using a int **bitmap instead of int *bitmap?
Additionally, why do we check a pointer (use_bitmap) and update another one?